src/relay.ts

```diff
@@ -66,32 +66,59 @@ export default async function relay(
     logger.info(`Relay succeeded: ${hash}`);
     return true;
   } catch (err) {
-    if (err instanceof BaseError) {
-      const revertError = err.walk(
-        (err) => err instanceof ContractFunctionRevertedError,
-      );
-      if (revertError instanceof ContractFunctionRevertedError) {
-        const errName = revertError.data?.errorName ?? "";
-        if (errName !== "Error") {
-          // One of the custom errors we defined, which should include
-          // a more detailed message in the metaMessages field
-          const metaMsg = revertError.metaMessages
-            ?.map((msg: string) => msg.trim())
-            .join("");
+    if (!(err instanceof BaseError)) {
+      // Theoretically should never happen, as all errors in Viem should extend BaseError
+      logger.error("Relay failed due to an unknown non-BaseError:", err);
+      return false;
+    }
 
-          logger.error("Contract reverted with error:", metaMsg);
-        } else {
-          // A generic revert error, which should include the reason
-          logger.error("Contract reverted with reason:", revertError.reason);
+    const revertError = err.walk(
+      (err) => err instanceof ContractFunctionRevertedError,
+    );
+    if (revertError instanceof ContractFunctionRevertedError) {
+      // If the error was a revert from the contract, it should fall into one of two types:
+      // 1. A custom error defined in the relayer contract, i.e. InvalidPrice, TimestampNotNew, etc.
+      // 2. An error from a require statement, in which case we try to extract the reason
+      const errName = revertError.data?.errorName ?? "";
+      switch (errName) {
+        case "TimestampNotNew": {
+          logger.info(
+            "Relay skipped. Price from previous relay is still fresh in SortedOracles",
+          );
+          break;
+        }
+        case "ExpiredTimestamp": {
+          logger.warn(
+            "Relay skipped. The current price is too old to be relayed",
```

Comment:
think we need a stronger word than **skipped** here, given it's a *warn* and not an *info*

either **Relay failed** (although that seems to be reserved for errors looks like) or **Relay not possible** ?

Reply:
Will change it to `Relay not possible`, as this is a log we would probably see relatively often from the time there's a successful relay until chainlink reports a new price